libvuln/driver/driver.go

```diff
@@ -73,3 +76,20 @@ type Fetcher interface {
 	// a sha265 sum of a OVAL xml file.
 	Fetch() (io.ReadCloser, string, error)
 }
+
+// FetcherNG is an experimental fetcher interface.
+//
+// This may go away or be renamed without warning.
+type FetcherNG interface {
+	FetchContext(context.Context, Fingerprint) (io.ReadCloser, Fingerprint, error)
+}
+
+// Unchanged is returned by Fetchers when the database has not changed.
+var Unchanged = errors.New("database contents unchanged")
+
+// Fingerprint is some identifiying information about a vulnerability database.
+type Fingerprint struct {
```

Comment:
In this fingerprint what is SHA256 a byte array of ? 

This is also in conflict with the current update_cursor record: https://github.com/quay/claircore/blob/master/internal/vulnstore/postgres/bootstrap.sql#L50

We should reconcile what gets persisted and subsequently looked up to understand if a vulnerability database has changed.

Reply:
It's the SHA256 of the fetched content.

And yeah, this should get persisted but the first step is figuring out what the Fetcher will use to prevent duplicating work.

Reply:
@hdonnay we cannot assume etags, what if sources are git repos?

Reply:
@hdonnay what I had in my head originally was for the Fetch method implementer to decide what opaque value they want to use understand if the database has changed, and we simply store this without knowing the details. This gives the implementer a large degree of freedom. 

If the implementer decides to use the etag, or a sha256 of the actual contents, do we need to be aware of this from a "caller of the interface" perspective ?

Reply:
Yeah it might not exist, that's why there's multiple members and in the experimental interface an argument for passing something into the fetcher.

Reply:
I guess what I'm struggling with is can't 

SHA256 []byte and ETag string be reduced to just string

string([]byte) == sha256

Reply:
I guess what I'm struggling with is can't

SHA256 []byte and ETag string be reduced to just string

string([]byte) == sha256

Reply:
Because one is an opportunistic way to avoid work and one is a strong checksum.

Reply:
thinking about it more, I don't feel strongly about what's in the fingerprint, just that we should have something like a fingerprint